sonic-net/sonic-buildimage
# [dhcp_server] Add dhcpservd to dhcp_server container

**Depends on: https://github.com/sonic-net/sonic-buildimage/pull/16548**

#### Why I did it
Part implementation of dhcp_server. HLD: https://github.com/sonic-net/SONiC/pull/1282.
Add dhcpservd to dhcp_server container.

##### Work item tracking
- Microsoft ADO **(number only)**: 25093520

#### How I did it
1. Add installing required pkg (psutil) in `Dockerfile`.
2. Add copying required file to container in `Dockerfile` (kea-dhcp related and dhcpservd related)
3. Add critical_process and supervisor config.
4. Add support for generating kea config (only in dhcpservd.py) and updating lease table (in dhcpservd.py and lease_update.sh)

#### How to verify it
1. Build image with setting `INCLUDE_DHCP_SERVER` to y and enabled dhcp_server feature after installed image, container start as expected.
2. Enter container and found that all processes defined in supervisor configuration running as expected.
3. Kill processes defined in critical_processes, container exist.



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
